light_curves/code_src/sample_selection.py

```diff
@@ -499,6 +525,29 @@ def get_sdss_sample(coords, labels, *, num=10, zmin=0, zmax=10, randomize_z=Fals
         print('SDSS Quasar: ' + str(num))
 
 
+def Ned_query_refcode_exceptions(paper_link):
+    """
+    Call Ned.query_refcode() with basic network exception handling.
+
+    Parameters
+    ----------
+    paper_link : str
+        ADS/NED reference code or paper identifier.
+
+    Returns
+    -------
+    result : object or None
+        Result from Ned.query_refcode, or None if a network error occurs.
+    """
+    try:
+        return Ned.query_refcode(paper_link)
+    except (ConnectionError, RequestException, TimeoutError):
+        print(
+            f"WARNING: encountered a ConnectionError error for paper "
+            f"{paper_link}. skipping."
+        )
```

Comment:
Is there any reasons these are printouts rather than actual warnings? I think it would be nicer to be the latter, so downstream users can do something about them.